language/stdlib/modules/validator_config.move

```diff
@@ -1,156 +1,183 @@
     use 0x0::Option;
     use 0x0::Transaction;
+    use 0x0::ValidatorOperatorConfig;
 
     struct Config {
         consensus_pubkey: vector<u8>,
         // TODO(philiphayes): restructure
+        //   1) remove validator_network_signing_pubkey
+        //   2) make validator_network_address[es] of type vector<vector<u8>>,
+        //   3) make fullnodes_network_address[es] of type vector<vector<u8>>,
+        //   4) remove validator_network_identity_pubkey
+        //   5) remove full_node_network_identity_pubkey
         validator_network_signing_pubkey: vector<u8>,
         validator_network_identity_pubkey: vector<u8>,
         validator_network_address: vector<u8>,
         full_node_network_identity_pubkey: vector<u8>,
         full_node_network_address: vector<u8>,
     }
 
-    // A current or prospective validator should publish one of these under their accounts.
-    // If delegation is enabled, only the delegated_account can change the config.
-    // If delegation is disabled, only the owner of this resource can change the config.
-    // The entity that can change the config is called a "validator operator".
-    // The owner of this resource can enable delegation and set delegated account.
-    // The owner can also disable delegation at any time.
     resource struct T {
-        config: Config,
-        delegated_account: Option::T<address>,
+        // the flag is set by the Association
+        is_certified: bool,
+        // set and rotated by the operator_account
+        config: Option::T<Config>,
+        operator_account: Option::T<address>,
     }
 
     // TODO(valerini): add events here
 
-    // Returns true if addr has a published ValidatorConfig::T resource
-    public fun has(addr: address): bool {
-        exists<T>(addr)
+    ///////////////////////////////////////////////////////////////////////////
+    // Validator setup methods
+    ///////////////////////////////////////////////////////////////////////////
+
+    // This function instantiates a
+    // ValidatorConfig resourse under validator_address
+    public fun initialize(validator_address: address) {
+        // TODO(valerini): Association::assert_sender_is_association();
+        //                 use move_to(...) to move resource to validator_address
+        // Currently callable by the validator
+        Transaction::assert(Transaction::sender() == validator_address, 1101);
+        move_to_sender<T>(T {
+            is_certified: false,
+            config: Option::none(),
+            operator_account: Option::none(),
+        });
     }
 
-    // Get Config
-    public fun get_config(addr: address): Config acquires T {
-        *&borrow_global<T>(addr).config
-    }
-
-    // Get consensus_pubkey from Config
-    public fun get_consensus_pubkey(config_ref: &Config): vector<u8> {
-        *&config_ref.consensus_pubkey
+    // This function sets is_certified flag to true
+    public fun certify(validator_address: address) acquires T {
```

Comment:
I thought the intent here was that the association has already created the account so it doesn't need to certify it. I guess this maps to the current behavior of add_validator?

Reply:
Right, the challenge here is that the Association cannot instantiate a certified resource at the moment of account creation at the validator's account (because the `move_to` function is not available in move yet). But eventually, the Association who creates the account will certify it in the same transaction indeed, add_validator now is callable by the Association as a work-around...